### PR TITLE
Adding Realtek driver r8169 from kernel 4.19.128

### LINKED ISSUE #403: No link with the standard kernel using the r8169 driver in the main xcp-ng kernel
Installing xcp-ng 8.1 on my ThinkPad L430 resulted in a missing carrier on the built-in Realtek NIC. 

lspci -k:
```
0c:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller (rev 07)
	Subsystem: Lenovo Device 21f7
	Kernel driver in use: r8169
	Kernel modules: r8169
```

ip a:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc pfifo_fast master ovs-system state DOWN group default qlen 1000
    link/ether 3c:97:0e:9f:07:46 brd ff:ff:ff:ff:ff:ff
3: ovs-system: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether 42:00:4c:b8:c2:42 brd ff:ff:ff:ff:ff:ff
4: xenbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ether 3c:97:0e:9f:07:46 brd ff:ff:ff:ff:ff:ff
    inet 192.168.2.33/24 brd 192.168.2.255 scope global xenbr0
       valid_lft forever preferred_lft forever
```
ethtool eth0:
```
Settings for eth0:
	Supported ports: [ TP MII ]
	Supported link modes:   10baseT/Half 10baseT/Full 
	                        100baseT/Half 100baseT/Full 
	                        1000baseT/Half 1000baseT/Full 
	Supported pause frame use: Symmetric Receive-only
	Supports auto-negotiation: Yes
	Supported FEC modes: Not reported
	Advertised link modes:  10baseT/Half 10baseT/Full 
	                        100baseT/Half 100baseT/Full 
	                        1000baseT/Half 1000baseT/Full 
	Advertised pause frame use: Symmetric Receive-only
	Advertised auto-negotiation: Yes
	Advertised FEC modes: Not reported
	Speed: Unknown!
	Duplex: Unknown! (255)
	Port: MII
	PHYAD: 0
	Transceiver: internal
	Auto-negotiation: on
	Supports Wake-on: pumbg
	Wake-on: g
	Current message level: 0x00000033 (51)
			       drv probe ifdown ifup
	Link detected: no
```
dmesg:
```
[    2.940877] libphy: r8169: probed
[    2.941401] r8169 0000:0b:00.0 eth0: RTL8168evl/8111evl, 3c:97:0e:9f:07:46, XID 2c900800, IRQ 56
[    2.941404] r8169 0000:0b:00.0 eth0: jumbo features [frames: 9200 bytes, tx checksumming: ko]
[    2.944414] ehci-pci 0000:00:1a.0: irq 16, io mem 0xf251a000
.....
[   16.278757] r8169 0000:0b:00.0 side-1013-eth0: renamed from eth0
.....
[   36.165620] r8169 0000:0b:00.0: invalid large VPD tag 7f at offset 0
[   36.658495] r8169 0000:0b:00.0 eth0: renamed from side-1013-eth0
[   37.160297] 8021q: 802.1Q VLAN Support v1.8
[   38.440397] openvswitch: Open vSwitch switching datapath
[   40.404132] device ovs-system entered promiscuous mode
[   40.481649] device xenbr0 entered promiscuous mode
[   40.603803] RTL8211E Gigabit Ethernet r8169-b00:00: attached PHY driver [RTL8211E Gigabit Ethernet] (mii_bus:phy_addr=r8169-b00:00, irq=IGNORE)
[   41.569480] device eth0 entered promiscuous mode
```

I have tried to set the interface to static during installation as well as DHCP (I have tried installing several times)

However installing and booting on the alternate kernel solved the problem
(https://xcp-ng.org/forum/topic/3199/problems-installing-latest-xcp-ng-on-a-thimkpad-l430)

### PR DESCRIPTION
**Pull request by rushikeshjadhav**
Fixes https://github.com/xcp-ng/xcp/issues/403

### CODE REVIEW

File: SPECS/r8169-module-alt.spec [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+%description
+%{vendor_name} %{driver_name} device drivers for the Linux Kernel
+version %{kernel_version}.
+
+%prep
+%autosetup -p1 -n %{driver_name}-%{version}
+
+%build
+%{?cov_wrap} %{make_build} -C /lib/modules/%{kernel_version}/build M=$(pwd) KSRC=/lib/modules/%{kernel_version}/build modules
```
Comment (stormi) [inline SPECS/r8169-module-alt.spec]: Do we need `%{?cov_wrap}`?

File: SPECS/r8169-module-alt.spec [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+%install
+%{?cov_wrap} %{__make} %{?_smp_mflags} -C /lib/modules/%{kernel_version}/build M=$(pwd) INSTALL_MOD_PATH=%{buildroot} INSTALL_MOD_DIR=%{module_dir} DEPMOD=/bin/true modules_install
```
Comment (stormi) [inline SPECS/r8169-module-alt.spec]: idem. And is the `_smp_mflags` macro available in our build environment or is it a Citrix thing?

File: SPECS/r8169-module-alt.spec (hunk 1 above)
Comment (rushikeshjadhav) [reply SPECS/r8169-module-alt.spec]: We can remove it. It's inherited from e1000e module.

File: SPECS/r8169-module-alt.spec (hunk 2 above)
Comment (rushikeshjadhav) [reply SPECS/r8169-module-alt.spec]: It is available and necessary to have parallel build.
```
# rpm --eval "Eval %{?_smp_mflags}"
Eval -j12
```

**Pull request closed without merging**